worker/repository/tasks.py

```diff
@@ -9,23 +9,28 @@
 staging_dir = os.environ['STAGING_DIR']
 
 
-@celery_app.task(bind=True, name="retrieve_from_staging", base=BaseTask)
-def task_retrieve_from_staging(self, object_id, job_id):
-    if not os.path.exists(staging_dir):
-        os.makedirs(staging_dir)
-    if not os.path.exists(working_dir):
-        os.makedirs(working_dir)
-    staging_path = os.path.join(staging_dir, object_id)
-    work_path = self.get_work_path(job_id)
-    if os.path.exists(work_path):
-        shutil.rmtree(work_path)
-    shutil.copytree(staging_path, work_path)
-
-
-@celery_app.task(bind=True, name="publish_to_repository", base=BaseTask)
-def task_publish_to_repository(self, object_id, job_id):
-    work_path = self.get_work_path(job_id)
-    repository_path = os.path.join(repository_dir, object_id)
-    if os.path.exists(repository_path):
-        shutil.rmtree(repository_path)
-    shutil.copytree(work_path, repository_path)
+class RetrieveFromStagingTask(BaseTask):
+
+    name = "retrieve_from_staging"
+
+    def execute_task(self):
+        staging_path = os.path.join(staging_dir, self.object_id)
+        work_path = self.get_work_path(self.job_id)
+        if os.path.exists(work_path):
+            shutil.rmtree(work_path)
+        shutil.copytree(staging_path, work_path)
+
+
+RetrieveFromStagingTask = celery_app.register_task(RetrieveFromStagingTask())
+
+
+class PublishToRepositoryTask(BaseTask):
```

Comment:
Mehrere Klassen in einem File bei uns/celery okay? Gehören ja zu einem Modul, also vermutlich schon?

Reply:
Okay, im dem Sinne, dass es funktioniert, ist es. Ich wollte jetzt nur nicht zu viel Refactoring auf einmal machen. Ich persönlich würde aber auch dafür plädieren, jeder Klasse ein eigenes File zu geben, könnte aber auch meine Java-Prägung sein. Können wir mal in der Runde diskutieren.

Reply:
Das ist klar, nur halt die Frage nach dem Stil. Aber dann bleibt es erst mal so und wir reden noch mal darüber, wie wir das mit zusammengehörigen Teams machen

Reply:
Also ich finde, dass man zusammengehörige und nicht zu lange Klassen ruhig in einer Datei haben kann. Habe zwar auch die Java Prägung, fand ich aber schon immer doof, dass man dazu gezwungen wird... :)